File: easy/3992_rearrange_string_to_avoid_character_pair/solution.cpp
#include <algorithm>
#include <string>

class Solution {
public:
  std::string rearrangeString(std::string s, char x, char y) {
    std::string t(std::count(s.begin(), s.end(), y), y);

    for (const char &ch : s) {
      if (ch != y)
        t.push_back(ch);
    }

    return t;
  }
};

File: easy/3992_rearrange_string_to_avoid_character_pair/solution.py
from collections import Counter


class Solution:
    def rearrangeString(self, s: str, x: str, y: str) -> str:
        sCounter = Counter(s)

        t = []

        for _ in range(sCounter.get(y, 0)):
            t.append(y)

        sCounter.pop(y, 0)

        for c, count in sCounter.items():
            for _ in range(count):
                t.append(c)

        return "".join(t)
